api_tests/nodes/views/test_node_detail.py

```diff
@@ -151,11 +152,28 @@ def test_node_has_files_link(self):
         assert_equal(urlparse(url).path, expected_url)
 
     def test_node_has_comments_link(self):
+        CommentFactory(node=self.public_project, user=self.user)
         res = self.app.get(self.public_url)
         assert_equal(res.status_code, 200)
         assert_in('comments', res.json['data']['relationships'].keys())
         url = res.json['data']['relationships']['comments']['links']['related']['href']
-        assert_equal(self.app.get(url).status_code, 200)
+        res = self.app.get(url)
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['data'][0]['type'], 'comments')
+
+    def test_node_comments_link_query_params_formattedg(self):
+        CommentFactory(node=self.public_project, user=self.user)
+        self.private_project_link = PrivateLinkFactory(anonymous=False)
+        self.private_project_link.nodes.append(self.private_project)
+        self.private_project_link.save()
+
+        res = self.app.get('{}?view_only={}'.format(self.private_url, self.private_project_link.key))
+        assert_equal(res.status_code, 200)
+        assert_in('comments', res.json['data']['relationships'].keys())
+        url = res.json['data']['relationships']['comments']['links']['related']['href']
+        res = self.app.get(url)
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['data'][0]['type'], 'comments')
```

Comment:
Could you verify in this test that the private link is included everywhere that private links are expected but weren't the first time through?